tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,245 @@
+        """
+        self.vault = pv_encryption_kms_setup_factory("v2", False)
+        self.sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.vault.kmsid,
+        )
+        self.vault.vault_path_token = self.vault.generate_vault_token()
+        self.vault.create_vault_csi_kms_token(namespace=self.proj_obj.namespace)
+        block_pvc_obj = pvc_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            storageclass=self.sc_obj,
+            size=3,
+            status=constants.STATUS_BOUND,
+            project=self.proj_obj,
+        )
+        pvc_clone_factory(block_pvc_obj)
+        snapshot_factory(block_pvc_obj)
+
+        """
+        ##################################### encrypted fs pvc
+        """
+        fs_pvc_obj = pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            storageclass=self.sc_obj,
+            size=3,
+            status=constants.STATUS_BOUND,
+            project=self.proj_obj,
+        )
+        pvc_clone_factory(fs_pvc_obj)
+        snapshot_factory(fs_pvc_obj)
+
+        """
+        ##################################### S3 buckets - TBD
+        """
+        # Create 1 bucket of each type (CLI, OC & S3)
+        self.bucket_names_list.append(bucket_factory(interface="CLI")[0].name)
+        self.bucket_names_list.append(bucket_factory(interface="OC")[0].name)
+        self.bucket_names_list.append(bucket_factory()[0].name)
+
+        obj_data = "A random string data"
+        key = "ObjKey-" + str(uuid.uuid4().hex)
+        assert s3_put_object(
+            mcg_obj, self.bucket_names_list[2], key, obj_data
+        ), f"Failed: Put object, {key}"
+        assert s3_get_object(
+            mcg_obj, self.bucket_names_list[2], key
+        ), f"Failed: Get object, {key}"
+
+        """
+        ##################################### mcg bucket replication      ---- TBD
+        """
+        # setup_mcg_system(2, 15)
+
+    @system_test
+    @polarion_id("OCS-3976")
+    @skipif_no_kms
+    def test_graceful_nodes_shutdown(
+        self, pgsql_factory_fixture, couchbase_factory_fixture, nodes
+    ):
+        """
+        Steps:
+          1) Have a cluster with FIPS, hugepages, encryption enabled
+          2) Create some resources: Create PVC (encrypt & non encrypt),
+            take snapshot and restore it into new PVCs, clone PVC
+          3) Configure OCP workloads(monitoring, registry, logging)
+          4) Configure PGSQL, Couchbase, AMQ , Cosbench, Jenkin, Quay workloads
+          5) Create rgw kafka notifications and objects should be notified
+          6) Create normal OBCs and buckets
+          7) Perform mcg bucket replication (uni and bidirectional) and see the objects are synced.
+          8) Perform noobaa caching
+          9) Perform S3 operations on NSFS bucket on files : put, get, delete, list objects
```

Comment:
Please add this also "10) With all the above preconditions met, follow the KCS steps for graceful shutdown of nodes https://access.redhat.com/articles/5394611, and bring up the cluster back.
Once cluster up, there should not be seen any issues or imapct on any data(No DU/DL/DC) and also normal operations should work fine"

Reply:
Done